codex-rs/tui/src/app.rs

```diff
@@ -162,17 +174,39 @@ impl<'a> App<'a> {
         self.app_event_tx.clone()
     }
 
+    /// Schedule a redraw if one is not already pending.
+    #[allow(clippy::unwrap_used)]
+    fn schedule_redraw(&self) {
+        let mut flag = self.pending_redraw.lock().unwrap();
+        if *flag {
+            return;
+        }
+        *flag = true;
+        let tx = self.app_event_tx.clone();
+        let pending = Arc::clone(&self.pending_redraw);
+        thread::spawn(move || {
+            thread::sleep(REDRAW_DEBOUNCE);
+            tx.send(AppEvent::Redraw);
+            #[allow(clippy::unwrap_used)]
+            let mut f = pending.lock().unwrap();
+            *f = false;
+        });
```

Comment:
This is slightly better because it results in holding the lock for a shorter amount of time.

That extra level of scoping around the use of `self.pending_redraw.lock()` and `flag` ensures that after `*flag = true`, the lock is dropped.

Also, `Arc::clone()` is less canonical than just invoking `.clone()`, in my experience.

```suggestion
        {
            #[allow(clippy::unwrap_used)]
            let mut flag = self.pending_redraw.lock().unwrap();
            if *flag {
                return;
            }
            *flag = true;
        }

        let tx = self.app_event_tx.clone();
        let pending_redraw = &self.pending_redraw.clone();
        thread::spawn(move || {
            thread::sleep(REDRAW_DEBOUNCE);
            tx.send(AppEvent::Redraw);
            #[allow(clippy::unwrap_used)]
            let mut f = pending.lock().unwrap();
            *f = false;
        });
```